mono/mini/mini-arm.c

```diff
@@ -2327,9 +2327,13 @@ mono_arch_get_llvm_call_info (MonoCompile *cfg, MonoMethodSignature *sig)
 	case RegTypeIRegPair:
 		break;
 	case RegTypeStructByAddr:
-		/* Vtype returned using a hidden argument */
-		linfo->ret.storage = LLVMArgVtypeRetAddr;
-		linfo->vret_arg_index = cinfo->vret_arg_index;
+		if (sig->pinvoke) {
+			linfo->ret.storage = LLVMArgVtypeByRef;
+		} else {
```

Comment:
Shouldn't this be done only for arm64_32 ?

Reply:
LLVM will translate it properly on armv7k as well

Reply:
What about normal arm linux ?

Reply:
I didn't test that configuration, but I don't see why LLVM shouldn't be able to emit proper code for that. It will emit whatever is right for the target ABI which is not always compatible with our internal ABI, thus the case distinction.

Reply:
Sometimes it can emit it, sometimes it can't, this is the reason for all the LLVMArg cases, the front end has to know ABI details in a lot of cases.

Reply:
Unfortunately, there is no arm+llvm CI lane.

Reply:
I'll have a look

Reply:
verified on armhf linux, generated code looks the same.